wire/wire.go

```diff
@@ -43,26 +48,32 @@ func (msg Msg) Marshal(buf []byte, rem int) ([]byte, int, error) {
 	if err != nil {
 		return buf, rem, fmt.Errorf("marshal type: %v", err)
 	}
-	buf, rem, err = surge.MarshalBytes(msg.Data, buf, rem)
+	buf, rem, err = surge.Marshal(msg.To, buf, rem)
 	if err != nil {
-		return buf, rem, fmt.Errorf("marshal data: %v", err)
+		return buf, rem, fmt.Errorf("marshal to: %v", err)
 	}
-	return buf, rem, nil
+	if rem < len(msg.Data) {
+		return buf, rem, fmt.Errorf("marshal to: insufficient buffer lenght")
+	}
+	copy(buf, msg.Data)
```

Comment:
`MarshalBytes` adds the length to the marshalled data, and the rust implementation only expects the data (sans the length). Ross recommended that I marshal it manually in order for it to be compatible with aw-rust

Reply:
This feels a little non-standard and error prone. Be aware that, when unmarshalling, `buf` will always need to be the perfect size. Otherwise, unmarshalling will fail.

Reply:
Gotcha, I'll change it